### PR TITLE
Replaced 'replicate' to correctly named 'replace' flags

### PR DESCRIPTION
**Pull request by jpien13**
**Renamed and replaced shouldReplicateSequentialMemories to shouldReplaceSequentialMemories per issue #7384**

**MLIR Tests**:
[0/1] Running the MLIR regression tests

Testing Time: 53.62s

Total Discovered Tests: 2877
  Unsupported      :  506 (17.59%)
  Passed           : 2370 (82.38%)
  Expectedly Failed:    1 (0.03%)

**CIRCT Tests**:
[0/1] Running the CIRCT regression tests

Testing Time: 12.50s

Total Discovered Tests: 824
  Unsupported      :  21 (2.55%)
  Passed           : 797 (96.72%)
  Expectedly Failed:   6 (0.73%)

**CIRCT integration tests**:
[1/2] Running the CIRCT integration tests

Testing Time: 15.29s

Total Discovered Tests: 98
  Unsupported      : 63 (64.29%)
  Passed           : 34 (34.69%)
  Expectedly Failed:  1 (1.02%)

### CODE REVIEW

**Comment by mwachs5** [comment]
@jpien13 you are failing the code formatting check. See https://github.com/llvm/circt/blob/bec0deab4bfa79785e3027bc95548435f58bd294/docs/GettingStarted.md?plain=1#L239. You can just do what the C/I check says (as I just learned myself as my PR is also currently failing the check, you can click the red X for your failing test, then scroll a few lines down below the failing job and expand to see the actual formatting issues --
Screenshot 2024-08-05 at 2 16 06 PM and try to fix them manually.

Or even better you can get clang-format working in your environment, the formatting configuration file is included in the repo.

File: lib/Firtool/Firtool.cpp [hunk 1]
```diff
@@ -78,7 +78,7 @@ LogicalResult firtool::populateCHIRRTLToLowFIRRTL(mlir::PassManager &pm,
   pm.nest<firrtl::CircuitOp>().addPass(firrtl::createInferWidthsPass());
 
   pm.nest<firrtl::CircuitOp>().addPass(
-      firrtl::createMemToRegOfVecPass(opt.shouldReplicateSequentialMemories(),
+      firrtl::createMemToRegOfVecPass(opt.shouldReplaceSequentialMemories(),
```
Comment (mwachs5) [inline lib/Firtool/Firtool.cpp:81]: the formatter doesn't like the trailing whitespace at the end of this line

Comment (jpien13) [reply lib/Firtool/Firtool.cpp]: Hi Megan! Yea i just saw this, i made the change just now. Had to install the clang formatter. Should be good now? We'll see what CI job says

Comment (jpien13) [reply lib/Firtool/Firtool.cpp]: I'll have to doublecheck the config file, I didn't see your comments till now (before I pushed the newest change to get rid of the whitespace). I'll check my settings and formatting now. Thanks :)

**Comment by mwachs5** [comment]
LGTM! @youngar can this be merged? I don't have permissions to do so.

**Comment by youngar** [comment]
Thanks @jpien13!

**Pull request merged**